test/e2e/operator_test.go

```diff
@@ -1842,8 +1842,9 @@ func TestIngressControllerCustomEndpoints(t *testing.T) {
 		}
 		defer func() {
 			// Remove the custom endpoints from the infrastructure config.
-			infraConfig.Spec.PlatformSpec.AWS = nil
-			if err := kclient.Update(context.TODO(), &infraConfig); err != nil {
+			if err := updateInfrastructureConfigSpecWithRetryOnConflict(t, types.NamespacedName{Name: "cluster"}, timeout, func(spec *configv1.InfrastructureSpec) {
+				spec.PlatformSpec.AWS = nil
+			}); err != nil {
 				t.Fatalf("failed to update infrastructure config: %v\n", err)
```

Comment:
The `\n` is unnecessary.  But you didn't add it in this PR, so we can address it in a follow-up.

Reply:
Again, left as-is to reduce the review footprint.